.github/workflows/release.yml

```diff
@@ -0,0 +1,119 @@
+name: Build & Release
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    name: Build
+    runs-on: ${{ matrix.build.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        build:
+          - {
+              NAME: linux-x64,
+              OS: ubuntu-22.04,
+              TARGET: x86_64-unknown-linux-gnu,
+            }
+          - {
+              NAME: windows-x64,
+              OS: windows-2022,
+              TARGET: x86_64-pc-windows-msvc,
+            }
+          - {
+              NAME: windows-arm64,
+              OS: windows-2022,
+              TARGET: aarch64-pc-windows-msvc,
+            }
+          - {
+              NAME: darwin-x64,
+              OS: macos-12,
+              TARGET: x86_64-apple-darwin,
+            }
+          - {
+              NAME: darwin-arm64,
+              OS: macos-12,
+              TARGET: aarch64-apple-darwin,
+            }
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Rust env
+        uses: "./.github/actions/setup-rust-env"
+
+      - name: Build
+        run: cargo build --release --locked --target ${{ matrix.build.TARGET }}
+
```

Comment:
Shall we add a step for tests also?

Reply:
Absolutely, but I think that would be best have have in a separate workflow (there also aren't any tests yet that I'm aware of) since this workflow is quite expensive (the windows and macos machines take quite a while).

My thoughts:
- Test: Build and Test just on a Linux machine (much faster) as I can't think of the benefit of running it on multiple machines
- Lint: Your standard rustfmt & clippy
- Release: this workflow as is that does a matrix build and potential release (perhaps also constraining it to only run on a push to master AND change to `Cargo.toml`). This workflow would be trigger on the completion of the above two, so tests and lint is passing

Whats your opinion on that setup?

Reply:
> Absolutely, but I think that would be best have have in a separate workflow (there also aren't any tests yet that I'm aware of) since this workflow is quite expensive (the windows and macos machines take quite a while).
> Test: Build and Test just on a Linux machine (much faster) as I can't think of the benefit of running it on multiple machines

Aha makes a ton of sense, I personally like having all the steps (test, lint, build, etc) before creating and uploading the artifacts but I agree it makes it super slow. Also makes sense to run the tests/lint in only one system, it won't impact them in any way.

I like the conditional running of the "slow" workflow for the quick feedback 👍